daphne_server/src/roles/aggregator.rs

```diff
@@ -184,6 +477,78 @@ impl HpkeDecrypter for crate::App {
         aad: &[u8],
         ciphertext: &HpkeCiphertext,
     ) -> Result<Vec<u8>, DapError> {
-        todo!()
+        let version = self
+            .get_task_config_for(task_id)
+            .await?
+            .as_ref()
+            .ok_or(DapAbort::UnrecognizedTask)?
+            .version;
+        self.kv()
+            .get_mapped::<kv::prefix::HpkeReceiverConfigSet, _, _>(&version, |config_list| {
+                config_list
+                    .iter()
+                    .find(|receiver| receiver.config.id == ciphertext.config_id)
+                    .map(|receiver| {
+                        receiver.decrypt(info, aad, &ciphertext.enc, &ciphertext.payload)
+                    })
+            })
+            .await
+            .map_err(|e| fatal_error!(err = ?e))?
+            .ok_or(DapError::Transition(TransitionFailure::HpkeUnknownConfigId))?
+    }
+}
+
+#[async_trait]
+impl BearerTokenProvider for crate::App {
```

Comment:
I think you already `use` this (if not, might as well?)
```suggestion
impl BearerTokenProvider for App {
```